src/caselawclient/models/identifiers/collection.py

```diff
@@ -22,20 +21,54 @@
 
 
 class IdentifiersCollection(dict[str, Identifier]):
-    def validate_uuids_match_keys(self) -> None:
+    def validate_uuids_match_keys(self) -> SuccessFailureMessageTuple:
         for uuid, identifier in self.items():
             if uuid != identifier.uuid:
-                msg = "Key of {identifier} in Identifiers is {uuid} not {identifier.uuid}"
-                raise UUIDMismatchError(msg)
+                return SuccessFailureMessageTuple(
```

Comment:
It's possible that having SuccessMessageTuple and FailureMessageTuple child classes would improve readability.